File: 14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        i = 0
        result = ''
        if strs[0] == "":
            return ""
        while i < len(strs[0]):
            item = strs[0][i]
            for s in strs:
                if i >= len(s):
                    return result
                # print(s[i])
                if s[i] != item:
                    return result
            result += item
            i += 1
        return result

File: Javascript/14. Longest Common Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs[0] == "") return "";

    let result = ""
    let i = 0
    while (i < strs[0].length) {
        let current = strs[0][i]
        for (str of strs) {
            if (i >= str.length || str[i] != current) {
                return result;
            }
        }
        result += current
        i += 1
    }
    return result
};
